### PR TITLE
passthrough_hp: include . and .. in readdir

### PR DESCRIPTION
**Pull request by joannekoong**
generic/401 fails currently because it checks that "." and ".." are listed as directory entries.

Include "." and ".." as listed directory entries in passthrough_hp's readdir implementation.

Signed-off by: Joanne Koong <joannelkoong@gmail.com>

### CODE REVIEW

File: example/passthrough_hp.cc [hunk 1]
```diff
@@ -747,15 +747,18 @@ static void do_readdir(fuse_req_t req, fuse_ino_t ino, size_t size,
...
 
         fuse_entry_param e{};
         size_t entsize;
         if (plus) {
-            err = do_lookup(ino, entry->d_name, &e);
-            if (err)
-                goto error;
+            if (is_dot_or_dotdot(entry->d_name)) {
+		e.attr.st_ino = entry->d_ino;
+		e.attr.st_mode = entry->d_type << 12;
```
Comment (bsbernd) [inline example/passthrough_hp.cc:756]: Would you mind to add a comment that fuse kernel does not refresh attributes and with that also does not increase nlookup? Without that comment I wasn't sure and had to look it up. Something like
/* fuse kernel ignores attributes for these and also does not increase lookup count */

**Comment by bsbernd** [comment]
Thank you, another issue fixed! Would be great if you could add a comment, so that one knows why there is an exception.

File: example/passthrough_hp.cc (hunk 1 above)
Comment (joannekoong) [reply example/passthrough_hp.cc:756]: Absolutely, I will add that in right now.

**Comment by bsbernd** [comment]
Thank you @joannekoong

**Comment by joannekoong** [comment]
Thanks for the review @bsbernd. 

I just reran the fuse fstests and this fixes generic/471 as well. Right now, I'm seeing about 25 failures (running with passthrough_hp). I have an upstream fix for generic/434 and generic/184, which I will submit this week and cc you on, Bernd.

**Pull request merged**